esp-config/src/generate.rs

```diff
@@ -0,0 +1,211 @@
+use core::fmt::Write;
+use std::{collections::HashMap, env, fs, path::PathBuf};
+
+const DOC_TABLE_HEADER: &str = r#"
+| Name | Description | Default value |
+|------|-------------|---------------|
+"#;
+const CHOSEN_TABLE_HEADER: &str = r#"
+| Name | Chosen value |
+|------|--------------|
+"#;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ParseError(String);
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Value {
+    Number(usize),
+    Bool(bool),
+    String(String),
+}
+
+impl Value {
+    fn parse(&mut self, s: &str) -> Result<(), ParseError> {
+        *self = match self {
+            Value::Bool(_) => match s {
+                "false" | "no" | "n" => Value::Bool(false),
+                "true" | "yes" | "y" => Value::Bool(true),
+                _ => return Err(ParseError(format!("Invalid boolean value: {}", s))),
+            },
+            Value::Number(_) => Value::Number(
+                match s.as_bytes() {
+                    [b'0', b'x', ..] => usize::from_str_radix(&s[2..], 16),
+                    [b'0', b'o', ..] => usize::from_str_radix(&s[2..], 8),
+                    [b'0', b'b', ..] => usize::from_str_radix(&s[2..], 2),
+                    _ => usize::from_str_radix(&s, 10),
+                }
+                .map_err(|_| ParseError(format!("Invalid numerical value: {}", s)))?,
+            ),
+            Value::String(_) => Value::String(String::from(s)),
+        };
+        Ok(())
+    }
+
+    fn as_string(&self) -> String {
+        match self {
+            Value::Bool(value) => String::from(if *value { "true" } else { "false" }),
+            Value::Number(value) => format!("{}", value),
+            Value::String(value) => value.clone(),
+        }
+    }
+}
+
+/// This function will parse any screaming snake case environment variables that
```

Comment:
```suggestion
/// This function will parse any `SCREAMING_SNAKE_CASE` environment variables that
```

Just to be absolutely unambiguous.